packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -486,125 +594,125 @@ export class RunningSummarizer implements IDisposable {
+                opsSinceLastSummary: referenceSequenceNumber - this.heuristics.lastAcked.refSequenceNumber,
+            };
+            this.logger.sendTelemetryEvent({
+                eventName: "GenerateSummary",
+                ...telemetryProps,
+            });
         } catch (error) {
-            summarizingEvent.cancel({ category: "error" }, error);
-            return;
+            return fail("generateSummaryFailure", "Error while generating or submitting summary", error);
+        } finally {
+            this.heuristics.recordAttempt(summaryData?.referenceSequenceNumber);
+            this.summarizeTimer.clear();
         }
 
-        this.summarizeTimer.clear();
-
-        if (!summaryData) {
-            summarizingEvent.cancel();
-            return;
+        if (!summaryData.submitted) {
+            // Did not send the summary op
+            return fail("generateSummaryFailure", "Error while generating or submitting summary", summaryData.error);
         }
 
-        const telemetryProps: any = {
-            ...summaryData,
-            ...summaryData.summaryStats,
-            refSequenceNumber: summaryData.referenceSequenceNumber,
-            opsSinceLastAttempt: summaryData.referenceSequenceNumber - this.heuristics.lastAttempted.refSequenceNumber,
-            opsSinceLastSummary: summaryData.referenceSequenceNumber - this.heuristics.lastAcked.refSequenceNumber,
-        };
-        telemetryProps.summaryStats = undefined;
-        telemetryProps.referenceSequenceNumber = undefined;
+        try {
+            const pendingTimeoutP = this.pendingAckTimer.start().catch(() => undefined);
+            const summary = this.summaryWatcher.watchSummary(summaryData.clientSequenceNumber);
 
-        if (summaryData.submitted) {
-            summarizingEvent.end(telemetryProps);
-        } else {
-            summarizingEvent.cancel(telemetryProps);
-        }
+            // Wait for broadcast
+            const summaryOp = await Promise.race([summary.waitBroadcast(), pendingTimeoutP]);
+            if (!checkNotTimeout(summaryOp)) {
+                return fail("summaryOpWaitTimeout", "Timeout while waiting for summarize op broadcast");
+            }
+            this.heuristics.lastAttempted.summarySequenceNumber = summaryOp.sequenceNumber;
+            this.logger.sendTelemetryEvent({
+                eventName: "SummaryOp",
+                timeWaiting: Date.now() - this.heuristics.lastAttempted.summaryTime,
+                refSequenceNumber: summaryOp.referenceSequenceNumber,
+                summarySequenceNumber: summaryOp.sequenceNumber,
+                handle: summaryOp.contents.handle,
+            });
 
-        return summaryData;
+            // Wait for ack/nack
+            const ackNack = await Promise.race([summary.waitAckNack(), pendingTimeoutP]);
+            if (!checkNotTimeout(ackNack)) {
+                return fail("summaryAckWaitTimeout", "Timeout while waiting for summaryAck/summaryNack op");
```

Comment:
Note: We will send 2 events and warning propagations for summary ack wait timeout now: 1 for the timer handler and 1 here. During the first load, only the ones for the timer handler will go off.